logger/logger.go

```diff
@@ -64,6 +64,10 @@ type logger struct {
 	Logger kitlog.Logger
 }
 
+func (l *logger) Decorate(keyvals ...interface{}) {
+	l.Logger = kitlog.NewContext(l.Logger).With(keyvals...)
```

Comment:
> This could return new logger instance.

Yes, but how do you use it then? AFAIK it is for decorating the logs with a cluster ID. Even with a new logger you have the thread safety problem. So you end up creating a new logger in each method where the cluster ID is being passed. I prefer just using the logger the service has configured and explicitly provide the information that have to be logged, but this is just me. 

> I think this should be possible.

Oh yeah, totally. The question is only how and I seem to miss the idea or the big picture. 

> Otherwise why should we inject the logger to every service instead using global one?

Because the design is just better. We can better test and maintain it that way. Also, having a global logger you have to create a new instance in each method as well to decorate it or you have to add the items to the logging explicitly as well.